spdlog warning under NDEBUG

A warning reported by TRI's downstream CI:
```
INFO: From Compiling external/drake/bindings/pydrake/common/text_logging_pybind.cc:
In file included from external/drake/bindings/pydrake/common/text_logging_pybind.cc:12:0:
bazel-out/k8-opt/bin/external/drake/tools/install/libdrake/_virtual_includes/drake_shared_library/drake/common/text_logging.h:39:0: warning: "SPDLOG_ACTIVE_LEVEL" redefined
 #define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_TRACE
 
In file included from external/spdlog/include/spdlog/sinks/base_sink.h:12:0,
                 from external/drake/bindings/pydrake/common/text_logging_pybind.cc:7:
external/spdlog/include/spdlog/common.h:134:0: note: this is the location of the previous definition
 #define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_INFO
```

This is with assertions (NDEBUG) enabled.

Best guess, include statements are in the wrong order.

Started as of https://github.com/RobotLocomotion/drake/pull/13568, I believe.

--- a/bindings/pydrake/common/text_logging_pybind.cc
+++ b/bindings/pydrake/common/text_logging_pybind.cc
@@ -4,12 +4,17 @@
 #include <memory>
 #include <mutex>
 
+// clang-format off to disable clang-format-includes
+// N.B. text-logging.h must be included before spdlog headers
+// to avoid "SPDLOG_ACTIVE_LEVEL" redefined warning (#13771).
+#include "drake/common/text_logging.h"
+// clang-format on
+
 #include <spdlog/sinks/base_sink.h>
 #include <spdlog/sinks/dist_sink.h>
 
 #include "drake/bindings/pydrake/pydrake_pybind.h"
 #include "drake/common/drake_assert.h"
-#include "drake/common/text_logging.h"
 #endif
 
 namespace drake {

